src/oal_agent/telemetry/metrics.py

```
@@ -29,5 +33,32 @@ def get_high_water_mark(self, metric: str) -> float:
+            g.set(value)
+
+        try:
+            push_to_gateway(
+                gateway=settings.prometheus_pushgateway_url,
+                job=settings.prometheus_pushgateway_job,
+                registry=registry,
+            )
+            print(
+                f"Metrics pushed to Prometheus Pushgateway at {settings.prometheus_pushgateway_url}"
+            )
+        except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)